### ISSUE TITLE
Compilation failure: OpenCV SIFT: `Allocate` has incorrect return type.

### ISSUE THREAD

**Issue by tomas789** [openMVG/openMVG#1096]
Build fails when including SIFT from OpenCV. I'm getting following error
```
[ 82%] Building CXX object software/SfM/CMakeFiles/openMVG_main_ComputeFeatures_OpenCV.dir/main_ComputeFeatures_OpenCV.cpp.o
/Users/tomaskrejci/Developer/openMVG/src/software/SfM/main_ComputeFeatures_OpenCV.cpp:137:33: error: virtual function 'Allocate' has a different return type ('unique_ptr<Regions_type>') than the function it overrides
      (which has return type 'unique_ptr<openMVG::features::Regions>')
  std::unique_ptr<Regions_type> Allocate() const override
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^
/Users/tomaskrejci/Developer/openMVG/src/openMVG/features/image_describer.hpp:76:36: note: overridden virtual function is here
  virtual std::unique_ptr<Regions> Allocate() const = 0;
          ~~~~~~~~~~~~~~~~~~~~~~~~ ^
```

After modifying return type of `Allocate` function to `std::unique_ptr<openMVG::features::Regions>` everything seems to build well.

**Comment by pmoulon** [comment]
We made some update recently and certainly we broke this.
Can you try on the develop branch and make a PR if necessary?

**Comment by pmoulon** [comment]
Fixed by #1098
